src/perform_step/low_storage_rk_perform_step.jl

```diff
@@ -52,18 +56,29 @@ end
   @.. u   = u + B1*tmp
   # other stages
   for i in eachindex(A2end)
-    if williamson_condition
-      f(ArrayFuse(tmp, u, (A2end[i], dt, B2end[i])), u, p, t+c2end[i]*dt)
-    else
-      @.. tmp = A2end[i]*tmp
-      f(k, u, p, t+c2end[i]*dt)
-      @.. tmp += dt * k
+    if f isa IncrementingODEFunction
+      @.. tmp = (A2end[i] / dt)*tmp
```

Comment:
how come it's required here but not in the other?

Reply:
https://github.com/SciML/OrdinaryDiffEq.jl/pull/1185/files#diff-32bcae2e479bdf2c77ea6287240cdcbcR62 we multiply dt again here. In increment form, the function does `du += f(u,p,t)` so to implement `du += k * f(u,p,t)` we need to divide du by k first, then pass it in the function them multiply it back with k

Reply:
is that still hitting the minimal number of kernel calls?

Reply:
Nope that's still 2 kernel calls. We can't make it to 1 without kernel abstractions